src/ch/blinkenlights/android/medialibrary/MediaScanner.java

```diff
@@ -487,13 +487,20 @@ private boolean rpcInspectFile(File file) {
 			if (discNumber == null)
 				discNumber = "1"; // untagged, but most likely '1' - this prevents annoying sorting issues with partially tagged files
 
-			long artistId = MediaLibrary.hash63(artist);
-			long albumId = MediaLibrary.hash63(album);
+			// albumartist is now always present
+			String albumartist = tags.getFirst(MediaMetadataExtractor.ALBUMARTIST);
+			if (albumartist == null)
+				albumartist = "<No Artist>";
+
+			long albumartistId = MediaLibrary.hash63(albumartist);
 
-			// Overwrite albumId with a hash that included the parent dir if set in preferences
+			// I have several albums all called "Greatest Hits" with different album artists
+			String albumIdString = album + "\n" + albumartist;
+			// Include the parent dir in the hash if set in preferences
 			if (prefs.groupAlbumsByFolder) {
-				albumId = MediaLibrary.hash63(album + "\n" + file.getParent());
+				albumIdString += "\n" + file.getParent();
```

Comment:
What is the intend of these changes? Making the AlbumArtist the 'primary' artist-source of an Album?
I think this should rather be a config option where the user can choose if he prefers artist-related actions on the artist or album-artist. These changes would probably cause issue with various corner cases (and would require a full library flush)

Reply:
> What is the intend of these changes? Making the AlbumArtist the 'primary' artist-source of an Album?

Yes.

> I think this should rather be a config option where the user can choose if he prefers artist-related actions on the artist or album-artist. 

I understand your point, but let's be realistic, can you imagine any real-life scenario where a user would prefer to use the "artist" field value that is different form the "album artist" value, to identify the album's artist and display that as the property of the album? I cannot imagine such an example, but if you can present one, I would like to learn about it.

Here is my rationale for why it would not make sense. The cases I can imagine where both "Artist" and "Album artist" fields are present in the song and are different are the following:
- the album is "an album of Artist1" and may include some of "featured artists" on some of the tracks.
- the album is "a compilation of a specific genre/type of music" performed by various artists. Then the "Album artist" value _should_ be "VA" or "Various Artists" and "Artist" values of individual songs could vary.
- the album is "an album of songs originally performed/authored by a band/composer/poet". In this case, the appropriate value for the "Album artist" field is most probably the name of that original author and individual songs' "Artist" values would vary.

As you can easily see, in all of those cases above, using the "artist" value for album's property always only creates confusion and never correctly achieves the goal of identifying the set of all the songs in this album as a proper album. This consideration makes me believe that this piece of logic should not be configurable but "Album artist" should always be used for identifying the sets of songs as albums. By the way, Google Play Music does this unconditionally too and that seems very reasonable.

I have nothing against a config option when it makes sense, but if having this option "off" would never make any sense for any real-life user, then we better not have the "off" option.

> These changes would probably cause issue with various corner cases 

Considering the recently introduced calculation of "Album artist" based on "Artist" when the former is absent, I don't really anticipate any such "corner cases".

> (and would require a full library flush)

Yes, I am not sure what is the right way to achieve this but I am sure there is an existing mechanism for such flushes and we should use it here.